app/iot/service.py

```diff
@@ -18,32 +18,32 @@ def connect(self) -> None:
     def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
```

Comment:
You are generating a random id for the device and assuming it will be unique. However, there is a small chance that the same id can be generated again. It would be better to check if the id already exists in the dictionary and generate a new one if it does.